caffe2/proto/torch.proto

```diff
@@ -23,6 +23,12 @@ message TensorDef {
   optional string device = 7;
 }
 
+message AttributeDef {
+  // The mypy type of this attribute
+  required string type = 1;
+  required string name = 2;
```

Comment:
Attributes are part of a module def (rather than the model def) so they should be nested within a module / submodule, the attribute itself doesn't need to know which module it's in.

Reply:
<in person> We still should have attribute_id. Currently there is an expectation that modules are traversed in the same order they are deserialized. There are many ways we might use a module in the future where this will not hold:

* We choose to deserialize only a submodule. 
* We do zip-file surgery to add a new submodule or reorder submodules.
* We allow something besides Modules to have module references.